Index: lesson_8/8_1_11.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>class Hero:\n\n    def __init__(self, name, motto):\n        self.name = name\n        self.motto = motto\n\n    def say(name, motto):\n        print(self.name, \"говорит:\", self.motto)\n# Не удаляйте код ниже, это проверка (иначе мы укусим вас за бочок)\n\npythomir = Hero(\"Питомир\", \"Кусь за Русь!\")\npythomir.say()\ndjangomir = Hero(\"Джангомир\", \"Работает - и ладно!\")\ndjangomir.say()\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lesson_8/8_1_11.py b/lesson_8/8_1_11.py
--- a/lesson_8/8_1_11.py	(revision a5debccc92dcb5985591f3668c26f3783780f0aa)
+++ b/lesson_8/8_1_11.py	(date 1651521958000)
@@ -4,8 +4,10 @@
         self.name = name
         self.motto = motto
 
-    def say(name, motto):
-        print(self.name, "говорит:", self.motto)
+    def say(self):
+        print(f'{self.name} говорит: {self.motto}')
+
+
 # Не удаляйте код ниже, это проверка (иначе мы укусим вас за бочок)
 
 pythomir = Hero("Питомир", "Кусь за Русь!")
Index: lesson_8/8_1_12.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>class Hero:\n\n  def __init__(self, name, things=[]):\n    self.name = name\n    self.things = things\n\n  def collect(self, things):\n    self.things.append(things)\n\n# Не удаляйте код ниже, это проверка (иначе мы укусим вас за бочок)\n\npythomir = Hero(\"Питомир\")\npythomir.collect(\"Усы хитрости\")\npythomir.collect(\"Рукавички пряморукости\")\n\nif len(pythomir.things) == 2:\n  print(\"Проверка пройдена\")\nelse:\n  print(\"Добавление в список things работает некорректно\")
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lesson_8/8_1_12.py b/lesson_8/8_1_12.py
--- a/lesson_8/8_1_12.py	(revision a5debccc92dcb5985591f3668c26f3783780f0aa)
+++ b/lesson_8/8_1_12.py	(date 1651521958000)
@@ -1,18 +1,20 @@
 class Hero:
 
-  def __init__(self, name, things=[]):
+  def __init__(self, name):
     self.name = name
-    self.things = things
+    self.things = []
 
   def collect(self, things):
     self.things.append(things)
 
+
 # Не удаляйте код ниже, это проверка (иначе мы укусим вас за бочок)
 
 pythomir = Hero("Питомир")
 pythomir.collect("Усы хитрости")
 pythomir.collect("Рукавички пряморукости")
 
+
 if len(pythomir.things) == 2:
   print("Проверка пройдена")
 else:
